raft/server.cc

```diff
@@ -1441,8 +1457,9 @@ future<> server_impl::wait_for_apply(index_t idx, abort_source* as) {
         // This will be signalled when read_idx is applied
         auto it = _awaited_indexes.emplace(idx, awaited_index{{}, {}});
         if (as) {
-            it->second.abort = as->subscribe([this, it] () noexcept {
-                it->second.promise.set_exception(request_aborted());
+            it->second.abort = as->subscribe([this, it, applied_idx = _applied_idx] noexcept {
```

Comment:
Why did you decide to copy `_applied_idx` and print the copied value? Now it's worse than before. Now it may print outdated apply index, from the moment of constructing this lambda, not from the moment where the exception is created...

Reply:
Reverted, there was a comment to add more details. what else can I add here?

Reply:
Done